### ISSUE TITLE
"Compile" button disappears from Editor when changing editor

### ISSUE THREAD

**Issue by SkaveRat** [dotBunny/CLionSourceCodeAccess#96]
When I change the sourcecode editor from VS to CLion, the "Compile" button disappears from the editor.

Version 4.18

**Comment by S-Marais** [comment]
yes this is because for some reason the change of the CLion source code editor modifies the Engine.ini file.
You might also notice that if you try now to create a new cpp project it will ask you ty download NullSourceCodeAccess and complains about a compiler.

The solution is to go to /PATH/TO/UNREAL/ENGINE/4.18/Engine/Saved/Config/Linux/EditorSettings.ini and find the line:
`PreferredAccessor=CLionSourceCodeAccessor`
and to change it to
`PreferredAccessor=NullSourceCodeAccessor`
I'm not sure what can be done to make it work properly again.

**Comment by reapazor** [comment]
I've gotta look more into this, specifically the Windows workflow. 
On Mac/Linux CLion uses UBT to build inside of itself.

**Comment by S-Marais** [comment]
@reapazor this happens on Linux and here we are talking about the compile button from the editor, not the compilation inside of CLion.
This happens when you set the Source Code editor and then close the editor. When you reopen it EditorSettings.ini has been changed. You therefor can't create new c++ project or see the compile button within the project editor.

**Comment by reapazor** [comment]
Ahh... I'll follow up with Epic on this one :)

**Comment by S-Marais** [comment]
So here is a bit of information. As I stated this error happens when you use UE 4.18 and under. Reason is that CLionSourceCodeAccess is a per project plugin. When you install the plugin on a per project you should not change the Editor Settings at all. So when you install the plugin just use the File drop down menu to generate CMakeLists.txt and open CLion but do not change Editor Settings to set CLion as you Source Code editor.

You will not face this problem starting with 4.19 since the plugin is now a per Engine plugin.

I think it's important to update the Wiki to explain this part.
